Sources/GRPCHTTP2Core/Server/GRPCServerStreamHandler.swift

```diff
@@ -97,11 +101,27 @@ extension GRPCServerStreamHandler {
           endStream: headers.endStream
         )
         switch action {
-        case .receivedMetadata(let metadata):
-          context.fireChannelRead(self.wrapInboundOut(.metadata(metadata)))
+        case .receivedMetadata(let metadata, let methodDescriptor):
+          if let methodDescriptor = methodDescriptor {
+            self.methodDescriptorPromise.succeed(methodDescriptor)
+            context.fireChannelRead(self.wrapInboundOut(.metadata(metadata)))
+          } else {
+            self.methodDescriptorPromise.fail(
+              RPCError(
+                code: .internalError,
+                message: "Method descriptor should have been present if we received metadata."
+              )
+            )
```

Comment:
IIUC we should never hit this path on the server, so can we `assertionFailure` here instead?